### PR TITLE
add promise to async calling

### PR DESCRIPTION
**Pull request by rohozhnikoff**
i work with node-github with my two pet projects
usually i use promises for async calling, because it's more useful for chaining, grouping, step-by-step code, debugging, abstract construction of queries to pseudo-pure function and other
that's why i improve my api-calls with promise-returning, based on es6-standart
want to share this

### CODE REVIEW

**Comment by rohozhnikoff** [comment]
discussion in comment https://github.com/mikedeboer/node-github/issues/205

File: index.js [hunk 1]
```diff
@@ -363,7 +367,20 @@ var Client = module.exports = function(config) {
                             return;
                         }
 
-                        api[section][funcName].call(api, msg, block, callback);
+                        // good description for standart: http://www.html5rocks.com/en/tutorials/es6/promises/
```
Comment (Pouja) [inline index.js:370]: standard*

**Comment by Pouja** [comment]
Why can't an user just use Q.ninvoke? Or any other wrapper like function of Q?

**Comment by rohozhnikoff** [comment]
what reason to use Q instead of existing solutions?

> Or any other wrapper like function of Q?

es6 Promise is it

**Comment by jamestalmage** [comment]
@mikedeboer see #262, I think it is the better solution.

**Comment by kaizensoze** [comment]
Holding off on a promise solution for now, but will probably go with #262 if any. Thanks for the PR though!

**Comment by gr2m** [comment]
Hey folks, we plan to drop support for the custom Promise option altogether. If that causes any trouble for you, could you please let me know at #622? 

Thanks :)

**Pull request closed without merging**